### PR TITLE
dev-python/click: bump to 7.0

### PR DESCRIPTION
**Pull request by hydrapolic**
- pypi version dist has Click-7.0.tar.gz name :-/
- Makefile got removed
- docs need a new package pallets-sphinx-themes and thus needs rekeywording
- docs can be built using py3 (removed https://github.com/gentoo/gentoo/commit/de00f8e224cded2e844f11892a9eaea4c8251d33)

````
< SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
---
> SRC_URI="https://github.com/pallets/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
16c16
< KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86"
---
> KEYWORDS="~amd64 ~x86"
19,20d18
< REQUIRED_USE="doc? ( || ( $(python_gen_useflags 'python2*') ) )"
< 
24,35c22,23
<       doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )"
< 
< PATCHES=(
<       # From upstream, can be removed in next release.
<       "${FILESDIR}/${PN}-6.7-fix-tests.patch"
<       # From upstream, can be removed in next release.
<       "${FILESDIR}/${PN}-6.7-support-sphinx-1.7.patch"
< )
< 
< pkg_setup() {
<       use doc && DISTUTILS_ALL_SUBPHASE_IMPLS=( 'python2*' )
< }
---
>       doc? ( dev-python/sphinx[${PYTHON_USEDEP}]
>               dev-python/pallets-sphinx-themes[${PYTHON_USEDEP}] )"
48c36
<       emake test
---
>       pytest -vv || die

````

### CODE REVIEW

File: dev-python/click/click-7.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+DESCRIPTION="A Python package for creating beautiful command line interfaces"
+SRC_URI="https://github.com/pallets/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="http://click.pocoo.org/ https://pypi.org/project/click/"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc examples test"
+
+REQUIRED_USE="doc? ( || ( $(python_gen_useflags 'python2*') ) )"
```
Comment (mgorny) [inline dev-python/click/click-7.0.ebuild]: Why oh why do we need it?

File: dev-python/click/click-7.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	sed -e "s/, 'sphinx.ext.intersphinx'//" -i docs/conf.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile_all() {
+	use doc && emake -C docs html
+}
+
+python_test() {
+	py.test || die
```
Comment (mgorny) [inline dev-python/click/click-7.0.ebuild]: `pytest -vv` (`py.test` name is deprecated)

File: dev-python/click/click-7.0.ebuild (hunk 1 above)
Comment (hydrapolic) [reply dev-python/click/click-7.0.ebuild]: Actually I don't understand why is it here :-/

File: dev-python/click/click-7.0.ebuild (hunk 2 above)
Comment (hydrapolic) [reply dev-python/click/click-7.0.ebuild]: Thanks, will rename.

File: dev-python/click/click-7.0.ebuild (hunk 1 above)
Comment (hydrapolic) [reply dev-python/click/click-7.0.ebuild]: Probably because only py2 allowed to build docs some ages ago?

Comment (hydrapolic) [reply dev-python/click/click-7.0.ebuild]: https://github.com/gentoo/gentoo/commit/de00f8e224cded2e844f11892a9eaea4c8251d33

**Comment by hydrapolic** [comment]
Thanks @mgorny,sure,will do.

**Pull request closed without merging**